drizzlepac/hlautils/se_source_generation.py

```diff
@@ -387,46 +412,119 @@ def write_catalog(seg_cat, catalog_filename, product="tdp"):
+        seg_table["ycentroid"].info.format = ".10f"
         seg_table["RA_icrs"].info.format = ".10f"
         seg_table["Dec_icrs"].info.format = ".10f"
-        seg_table["RA_icrs"].unit = u.deg
-        seg_table["Dec_icrs"].unit =  u.deg
         print("seg_table (filter): {}".format(seg_table))
 
         seg_table.write(catalog_filename, format="ascii.ecsv")
         log.info("Wrote filter source catalog: {}".format(catalog_filename))
 
+def get_header_data(imghdu):
+
+    keyword_dict = {}
+
+    # Get the HSTWCS object from the primary header
+    keyword_dict["proposal_id"] = imghdu[0].header["PROPOSID"]
+    keyword_dict["image_file_name"] = imghdu[0].header['FILENAME']
+    keyword_dict["target_name"] = imghdu[0].header["TARGNAME"] 
+    keyword_dict["date_obs"] = imghdu[0].header["DATE-OBS"]
+    keyword_dict["instrument"] = imghdu[0].header["INSTRUME"]
+    keyword_dict["detector"] = imghdu[0].header["DETECTOR"]
+    keyword_dict["target_ra"] = imghdu[0].header["RA_TARG"]
+    keyword_dict["target_dec"] = imghdu[0].header["DEC_TARG"]
+    keyword_dict["expo_start"] = imghdu[0].header["EXPSTART"]
+    keyword_dict["expo_time"] = imghdu[0].header["EXPTIME"]
+    keyword_dict["ccd_gain"] = imghdu[0].header["CCDGAIN"]
+    keyword_dict["filter_tdp"] = imghdu[0].header["FILTER"]
+    keyword_dict["filter1"] = imghdu[0].header["FILTER1"]
+    keyword_dict["filter2"] = imghdu[0].header["FILTER2"]
+
+    # Get the HSTWCS object from the first extension
+    keyword_dict["wcs_name"] = imghdu[1].header["WCSNAME"]
+    keyword_dict["orientation"] = imghdu[1].header["ORIENTAT"]
+    keyword_dict["aperture_ra"] = imghdu[1].header["RA_APER"]
+    keyword_dict["aperture_dec"] = imghdu[1].header["DEC_APER"]
+    keyword_dict["aperture_pa"] = imghdu[1].header[" "]
+
+    return(keyword_dict)
+
+def annotate_table(input_table, keyword_dict, product="tdp"):
+
+    input_table.meta["WCSNAME"] = keyword_dict["wcs_name"]
+    input_table.meta["Proposal ID"] = keyword_dict["proposal_id"]
+    input_table.meta["Image File Name"] = keyword_dict['image_file_name']
+    input_table.meta["Target Name"] = keyword_dict["target_name"] 
+    input_table.meta["Date Observed"] = keyword_dict["date_obs"]
+    # FIX 
+    if product.lower() == "tdp":
+        input_table.meta["Time Observed"] = " "
+    else:
+        input_table.meta["Time Observed"] = "FIX ME"
+    input_table.meta["Instrument"] = keyword_dict["instrument"]
+    input_table.meta["Detector"] = keyword_dict["detector"]
+    input_table.meta["Target RA"] = keyword_dict["target_ra"]
+    input_table.meta["Target DEC"] = keyword_dict["target_dec"]
+    input_table.meta["Orientation"] = keyword_dict["orientation"]
+    input_table.meta["Aperture RA"] = keyword_dict["aperture_ra"]
+    input_table.meta["Aperture DEC"] = keyword_dict["aperture_dec"]
+    input_table.meta["Aperture PA"] = keyword_dict["aperture_pa"]
+    if product.lower() == "tdp":
+        input_table.meta["Exposure Start"] = 0.000000 
```

Comment:
The exposure start time and exposure time for the total product can be obtained from the drizzle product header as a correctly computed values (expstart and texptime, respectively).  I am not sure why they are set to 0.0 here.  The total exposure time can be useful in interpreting the flux values.

Reply:
I can see the values in the white light, total filter, and individual filter FITS headers.  I put the EXPSTART and EXPTIME (or TEXPTIME)  into the total filter detection product catalog to match the example I pulled from MAST.  The "white light" catalog has these values set to 0.0 for some reason. I agree that each TDP should have the values and not zero.  I will fix.